File: sdks/csharp/src/NovaBilling/Types/ChurnCohortsResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record ChurnCohortsResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("months")]
    public IEnumerable<string> Months { get; set; } = new List<string>();

    [JsonPropertyName("cohorts")]
    public IEnumerable<CohortRow> Cohorts { get; set; } = new List<CohortRow>();

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/ruby/lib/novabilling/types/churn_cohorts_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class ChurnCohortsResponse < Internal::Types::Model
      field :months, -> { Internal::Types::Array[String] }, optional: false, nullable: false
      field :cohorts, -> { Internal::Types::Array[Novabilling::Types::CohortRow] }, optional: false, nullable: false
    end
  end
end
